api/controller.py

```diff
@@ -299,7 +300,9 @@ def get_domain(url):
 
         self.patron_web_domains = patron_web_domains
         self.setup_configuration_dependent_controllers()
-        self.authentication_for_opds_documents = {}
+        self.authentication_for_opds_documents = ExpiringDict(
+            max_len=1000, max_age_seconds=3600
+        )
```

Comment:
To fix [SIMPLY-2790](https://jira.nypl.org/browse/SIMPLY-2790) we want the dictionary of Authentication For OPDS to forget about any documents added to it more than an hour ago.

I don't think there should be a limit on the number of A4OPDS documents in the cache, but ExpiringDict doesn't work unless you specify `max_len`, so I set it to a very large number of libraries. I've asked Jonathan Green what is the largest number of libraries he'd put on a single circulation manager, and if that's more than 500 I'll set this to about twice whatever value he gives.